Data/models/FaviconMO.swift

```diff
@@ -15,16 +15,18 @@ public final class FaviconMO: NSManagedObject, CRUD {
 
     // MARK: - Public interface
     
-    public class func add(_ favicon: Favicon, forSiteUrl siteUrl: URL) {
-        DataController.perform { context in
+    public class func add(_ favicon: Favicon, forSiteUrl siteUrl: URL, persistent: Bool) {
+        DataController.perform(context: .new(inMemory: !persistent)) { context in
             var item = FaviconMO.get(forFaviconUrl: favicon.url, context: context)
             if item == nil {
                 item = FaviconMO(entity: FaviconMO.entity(context), insertInto: context)
                 item!.url = favicon.url
             }
             if item?.domain == nil {
+                let strategy: Domain.SaveStrategy = persistent ?.delayedPersistentStore : .inMemory
+                
                 item!.domain = Domain.getOrCreateInternal(siteUrl, context: context,
-                                                          saveStrategy: .delayedPersistentStore)
+                                                          saveStrategy: strategy)
```

Comment:
@iccub - The `in-memory` db is not cleared when a user exits PBM, so favicons for sites browsed in private browsing mode will be available in the normal browsing mode. Let me know if I'm missing something.

Reply:
We never user in-memory db for anything in normal browsing mode.

You are right with not clearing the in-memory on exit, I thought we do it, it is cleared on app relaunch

Reply:
Clearing the `in-memory` data on transitioning from normal to PBM makes sense. If not, users will be able to see the browsing history on re-transitioning to the PBM.

Reply:
Follow-up issue: https://github.com/brave/brave-ios/issues/2656